barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini.hpp

```diff
@@ -439,5 +439,71 @@ template <typename Curve> class ShpleminiVerifier_ {
             commitments.emplace_back(std::move(fold_commitments[j]));
         }
     }
+
+    /**
+     * @brief Combines scalars of repeating commitments to reduce the number of scalar multiplications performed by the
+     * verifier.
+     *
+     * @details The Shplemini verifier gets the access to multiple groups of commitments, some of which are duplicated
+     * by design. This method combines the scalars associated with these repeating commitments, reducing the total
+     * number of scalar multiplications required during verification.
+     * More specifically, the Shplemini verifier receives two or three groups of commitments: get_unshifted() and
+     * get_to_be_shifted() in the case of Ultra, Mega, and ECCVM Flavors; and get_unshifted_without_concatenated(),
+     * get_to_be_shifted(), and get_groups_to_be_concatenated() in the case of the TranslatorFlavor. The commitments are
+     * then placed in this specific order in a BatchOpeningClaim object containing a vector of commitments and a vector
+     * of scalars. This method iterates over the commitments and scalars in a BatchOpeningClaim object
+     *
+     */
+    static void remove_shifted_commitments(BatchOpeningClaim<Curve>& accumulator,
```

Comment:
I think this method should probably not be called from the verifier classes, but rather be part of the "reduce_verify" stage in Shplemini, in which a BatchOpeningClaim is computed and then the shifted commitments are removed.

Reply:
done